Sources/Networking/AlamofireNetworkClient/AlamofireNetworkClient.swift

```diff
@@ -91,6 +92,33 @@ public extension AlamofireNetworkClient {
     return .init(with: request)
   }
   
+  func request<E: Encodable>(
+    method: HTTPMethod,
+    endpoint: URLConvertible,
+    headers: HTTPHeaders? = nil,
+    encode: E,
+    encoder: JSONEncoder? = nil,
+    interceptor: RequestInterceptor? = nil) -> Request
+  {
```

Comment:
@tonikocjan I've decided to go with `JSONEncoder` vs configuration here. We are doing the same on decoding end.